File: Project2/Antelope.cpp
#include "Antelope.h"
#include "World.h"
#include <stdlib.h>

Antelope::Antelope(const Vector2& pos) : Animal(pos, 4, 4, 3, 'A')
{
}

Antelope::~Antelope()
{
}

void Antelope::Hit(Organism* attacker)
{
	if (rand() % 2 == 0 && world->HasEmptyNeighbor(pos))
	{
		MoveTo(world->GetEmptyNeighbor(pos));
		return;
	}
	Animal::Hit(attacker);
}

void Antelope::Movement()
{
	Vector2 target = pos;
	for (int i = 0; i < motionRange; i++)
	{
		target = target.GetNeighbor(rand() % 4);
	}
	MoveTo(target);
}

Antelope* Antelope::Clone(const Vector2& pos) const
{
	return new Antelope(pos);
}

std::string Antelope::GetName() const
{
	return "Antelope";
}

File: Project2/WorldSimulator/Animals/Antelope.java
package WorldSimulator.Animals;

import WorldSimulator.Animal;
import WorldSimulator.Organism;
import WorldSimulator.Vector2;

import java.awt.*;
import java.util.Random;

public class Antelope extends Animal
{
	final int motionRange = 2;

    public Antelope(Vector2 pos)
    {
        super(pos, 4, 4, 3, 'A', new Color(219, 127, 62));
    }

    @Override
    protected void Hit(Organism attacker)
    {
        Random rand = new Random();
        if (rand.nextBoolean() && world.HasEmptyNeighbor(pos))
        {
            MoveTo(world.GetEmptyNeighbor(pos));
            return;
        }
        super.Hit(attacker);
    }

    @Override
    protected void Movement()
    {
        Random rand = new Random();
        Vector2 target = pos;
        for (int i = 0; i < motionRange; i++)
        {
            target = target.GetNeighbor(rand.nextInt(4));
        }
        MoveTo(target);
    }

    @Override
    public String toString()
    {
        return "Antelope";
    }
}
